autonomy/configurations/base.py

```diff
@@ -248,10 +241,10 @@ def process_metadata(
             ),
         )
         _ = configuration.pop("extra", {})
-        overide_type = OverrideType(
-            configuration.pop("override_type", OverrideType.SINGULAR.value)
+        has_multiple_overrides = all(
+            map(lambda x: isinstance(x, int), configuration.keys())
```

Comment:
Assuming all of the keys will be integers eg (0, 1, ...) we will use this check to decide if the override is singular or multiple type